File: day_21/Sort_Array_by_Parity.js
/**
 * @param {number[]} A
 * @return {number[]}
 */
var sortArrayByParity = function(A) {
    let even = [],
        odd = [];
    for (const num of A){
        if (num % 2) odd.push(num);
        else even.push(num);
    }
    return even.concat(odd);
};

File: day_21/Sort_Array_By_Parity.py
from collections import deque


class Solution:
    def sortArrayByParity(self, A: [int]) -> [int]:
        parity_d = deque()
        for num in A:
            if num % 2:
                # odd number
                parity_d.append(num)
            else:
                parity_d.appendleft(num)
        return parity_d


